src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/PKCS1MaskGenerationMethod.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.Cryptography;
+
+namespace System.Security.Cryptography
+{
+    public class PKCS1MaskGenerationMethod : MaskGenerationMethod
+    {
+        private string _hashNameValue;
+        
+        public PKCS1MaskGenerationMethod()
+        {
+            _hashNameValue = "SHA1";
+        }
+
+        public string HashName
+        {
+            get { return _hashNameValue; }
+            set 
+            { 
+                if (value != null && value != "SHA1")
+                {
+                    throw new PlatformNotSupportedException();
+                }
+
+                _hashNameValue = "SHA1";
```

Comment:
This line isn't necessary.  The field is initialized to "SHA1", and we throw if it's set to anything other than SHA1 (ignoring null), so we don't need to set it again.  Then the field can be made readonly.